client/my-sites/email/inbox/index.js

```diff
@@ -33,26 +34,31 @@ const InboxManagement = ( { domains } ) => {
 		);
 	}
 
-	const domainsWithSubscriptions = domains.filter(
-		( domain ) => ! domain.isWPCOMDomain && hasPaidEmailWithUs( domain )
-	);
+	// Find out if the site has at least one mailbox in the site
+	const hasOneOrMoreMailboxes = domains
+		.filter( ( domain ) => ! domain.isWPCOMDomain )
+		.reduce( ( accumulator, domain ) => {
+			const mailboxCountForDomain =
+				getGSuiteMailboxCount( domain ) + getConfiguredTitanMailboxCount( domain );
+			accumulator = mailboxCountForDomain > 0;
```

Comment:
Might it be simpler to just use `.some()` here? We only need to find one relevant domain.
```suggestion
	const hasAnyMailboxes = domains
		.some( ( domain ) =>
			! domain.isWPCOMDomain &&
				// it may be worth also checking for has(EmailProduct)WithUs( domain ) for each branch below
				( ( getGSuiteMailboxCount( domain ) > 0 ) || ( getConfiguredTitanMailboxCount( domain ) > 0 );
```